File: Python/504. Base7.py
#Function convertToBase7 converts number num to string representing this number in base 7.
class Solution(object):
    def convertToBase7(self, num):
        if num == 0:
            return "0"
        digitsList = []
        sign = num//abs(num)
        num = abs(num)
        while num > 0:
            digitsList += [str(num % 7)]
            num //= 7
        digitsList.reverse()
        numberBase7 = "".join(digitsList)
        if sign == -1:
            return "-" + numberBase7
        return numberBase7

File: C++/504. Base 7.cpp
class Solution {
public:
    string convertToBase7(int num) {
        //initialization
        int n=0,x,y;
        string s="";
        y=num;
        //Base case
        if(num==0){s+="0";return s;}
        //Dividing a number unless it is reduced to zero
        while(num!=0){
            x=abs(num%7);
            num=num/7;
            s+=(x+48);
        }
        if(y<0)s=s+"-";
        //reversing the number string
        reverse(s.begin(), s.end()); 
        return s;
    }
};
